pkg/controller/deployment/deployment_controller.go

```diff
@@ -84,8 +91,63 @@ func (d *DeploymentController) reconcileDeployment(deployment *extensions.Deploy
 }
 
 func (d *DeploymentController) reconcileRecreateDeployment(deployment extensions.Deployment) error {
-	// TODO: implement me.
-	return nil
+	oldRCs, err := d.getOldRCs(deployment)
+	if err != nil {
+		return err
+	}
+
+	// kill all existing pods for old rcs
+	waitScale := sync.WaitGroup{}
+	waitScale.Add(len(oldRCs))
+	for _, oldRC := range oldRCs {
+		go func(rc *api.ReplicationController) {
+			defer waitScale.Done()
+			if _, err := d.scaleRCAndRecordEvent(rc, 0, deployment); err != nil {
+				glog.Errorf("Fail to scale replication controller: %q/%q", rc.Namespace, rc.Name)
+				util.HandleError(err)
+			}
+		}(oldRC)
+	}
+	waitScale.Wait()
```

Comment:
I think you need to test and bail out if an error occurred.  In particular, if you failed to scale, you probably shouldn't wait for the replicas to go to zero.